test/unit/cmap/connect.test.ts

```diff
@@ -185,9 +185,100 @@ describe('Connect Tests', function () {
     expect(error).to.be.instanceOf(MongoNetworkError);
   });
 
-  context('prepareHandshakeDocument', () => {
+  describe('prepareHandshakeDocument', () => {
+    describe('client environment (containers and FAAS)', () => {
+      const cachedEnv = process.env;
+
+      context('when only kubernetes is present', () => {
+        const authContext = {
+          connection: {},
+          options: { ...CONNECT_DEFAULTS }
+        };
+
+        beforeEach(() => {
+          process.env.KUBERNETES_SERVICE_HOST = 'I exist';
+        });
+
+        afterEach(() => {
+          process.env.KUBERNETES_SERVICE_HOST = cachedEnv.KUBERNETES_SERVICE_HOST;
```

Comment:
This will not restore `KUBERNETES_SERVICE_HOST` to what it was before if it was `undefined` (the most common case), it will set it to `'undefined'` (string) because every assignment to process.env stringifies the value.